ambry-api/src/main/java/com.github.ambry/account/Account.java

```diff
@@ -0,0 +1,263 @@
+    this.name = name;
+    this.status = status;
+    metadata = new JSONObject();
+    metadata.put(ACCOUNT_METADATA_VERSION_KEY, ACCOUNT_METADATA_VERSION_1);
+    metadata.put(ACCOUNT_ID_KEY, id);
+    metadata.put(ACCOUNT_NAME_KEY, name);
+    metadata.put(ACCOUNT_STATUS_KEY, status);
+    metadata.put(CONTAINERS_KEY, new JSONArray());
+  }
+
+  /**
+   * Gets the id of the account.
+   * @return The id of the account.
+   */
+  public short getId() {
+    return id;
+  }
+
+  /**
+   * Gets the name of the account.
+   * @return The name of the account.
+   */
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * Gets the metadata of the account in JSON format.
+   * @return The metadata of the account.
+   */
+  public JSONObject getMetadata() {
+    return metadata;
+  }
+
+  /**
+   * Gets the status of the account.
+   * @return The status of the account.
+   */
+  public String getStatus() {
+    return status;
+  }
+
+  /**
+   * Gets the {@link Container} of this account with the specified container id.
+   * @param containerId The id of the container to get.
+   * @return The {@link Container} of this account with the specified id, or {@code null} if such a
+   *                    container does not exist.
+   */
+  public Container getContainerByContainerId(short containerId) {
+    return containerIdToContainerMap.get(containerId);
+  }
+
+  /**
+   * Gets the {@link Container} of this account with the specified container name.
+   * @param containerName The name of the container to get.
+   * @return The {@link Container} of this account with the specified name, or {@code null} if such a
+   *                    container does not exist.
+   */
+  public Container getContainerByContainerName(String containerName) {
+    return containerNameToContainerMap.get(containerName);
+  }
+
+  /**
+   * Gets all the containers of this account in a list.
+   * @return All the containers of this account.
+   */
+  public List<Container> getAllContainers() {
+    return Collections.unmodifiableList(new ArrayList<>(containerIdToContainerMap.values()));
+  }
+
+  /**
+   * Adds a container to this account without modifying the metadata of the account.
+   * @param container The container to this account.
+   */
+  private void addContainer(Container container) {
+    containerIdToContainerMap.put(container.getId(), container);
+    containerNameToContainerMap.put(container.getName(), container);
+  }
+
+  /**
+   * Adds a {@link Container} and updates the corresponding container field in the account metadata. It will replace
+   * the container metadata if it already exists.
+   * @param container The container to add.
+   * @throws JSONException
+   */
+  public void addContainerAndMetadata(Container container) throws JSONException {
+    if (container == null) {
+      throw new IllegalArgumentException("Container cannot be null.");
+    }
+    short accountMetadataVersion = (short) metadata.getInt(ACCOUNT_METADATA_VERSION_KEY);
```

Comment:
why do we need this check here. An account can be created only by either of two constructors
Account(JSONObject metadata)
public Account(short id, String name, String status)
in both cases, only valid versions will results in creation of account object. Can you clarify on this